src/dotnet/MsbuildProject.cs

```diff
@@ -16,44 +19,49 @@ internal class MsbuildProject
     {
         const string ProjectItemElementType = "ProjectReference";
 
-        public ProjectRootElement Project { get; private set; }
+        public ProjectRootElement ProjectRoot { get; private set; }
         public string ProjectDirectory { get; private set; }
 
-        private MsbuildProject(ProjectRootElement project)
+        private ProjectCollection _collection;
+        private List<NuGetFramework> _cachedTfms = null;
+        private Project _cachedEvaluatedProject = null;
+
+        private MsbuildProject(ProjectCollection collection, ProjectRootElement project)
```

Comment:
@rainersigwald can you take a look at these changes? Should we really be using our own ProjectCollection or should we use the GlobalCollection?